PSAlphaFS.psm1

```diff
@@ -525,20 +525,44 @@ function Copy-LongItem
 		}
 			
 		$PathObject = New-Object -TypeName Alphaleonis.Win32.Filesystem.FileInfo -ArgumentList $Path, $PathFSFormatObject::FullPath
+
+        if ($Destination -ne $null) {
+          $Destination.TrimEnd()
+        }
         
 		$basename = $PathFSObject::GetFileName($Path)
 		$dBasename = $PathFSObject::GetFileName($Destination)
 		$dParent = $PathFSObject::GetDirectoryName($Destination)
         
-		$dBasenameObj = New-Object -TypeName Alphaleonis.Win32.Filesystem.FileInfo -ArgumentList $dBasename
-		$isFile = if($PathFSObject::HasExtension($dBasename) )
+		
+		$isFile = if($PathFSObject::HasExtension($dBasename))
 		{
+            #has an extention so destination is a file
+            #Question: how does it handle directory with a period in the name?
```

Comment:
you are right...the code dosent handle dirs. with a period in them because I was still thinking about how to reliably detect file from a folder. One idea was to add an 'itemtype' parameter like 'new-item' which would remove all ambiguity but needs some work.